packages/block-editor/src/layouts/grid.js

```diff
@@ -0,0 +1,170 @@
+		blockName,
+		hasBlockGapSupport,
+		layoutDefinitions,
+	} ) {
+		const { minimumColumnWidth = '12rem' } = layout;
+
+		// If a block's block.json skips serialization for spacing or spacing.blockGap,
+		// don't apply the user-defined value to the styles.
+		const blockGapValue =
+			style?.spacing?.blockGap &&
+			! shouldSkipSerialization( blockName, 'spacing', 'blockGap' )
+				? getGapCSSValue( style?.spacing?.blockGap, '0.5em' )
+				: undefined;
+
+		let output = '';
+		const rules = [];
+
+		if ( minimumColumnWidth ) {
+			rules.push(
+				`grid-template-columns: repeat(auto-fill, minmax(min(${ minimumColumnWidth }, 100%), 1fr))`
+			);
+		}
+
+		if ( rules.length ) {
+			output = `${ appendSelectors( selector ) } {
+				${ rules.join( '; ' ) };
+			}`;
+		}
+
+		// Output blockGap styles based on rules contained in layout definitions in theme.json.
+		if ( hasBlockGapSupport && blockGapValue ) {
+			output += getBlockGapCSS(
+				selector,
+				layoutDefinitions,
+				'grid',
+				blockGapValue
+			);
+		}
+		return output;
+	},
+	getOrientation() {
+		return 'horizontal';
+	},
+	getAlignments() {
+		return [];
+	},
+};
+
+// Enables setting minimum width of grid items.
+function GridLayoutMinimumWidthControl( { layout, onChange } ) {
+	const { minimumColumnWidth: value = '12rem' } = layout;
+	const [ quantity, unit ] = parseQuantityAndUnitFromRawValue( value );
+
+	const handleSliderChange = ( next ) => {
+		onChange( {
+			...layout,
+			minimumColumnWidth: [ next, unit ].join( '' ),
+		} );
+	};
+
+	// Mostly copied from HeightControl.
+	const handleUnitChange = ( newUnit ) => {
+		// Attempt to smooth over differences between currentUnit and newUnit.
+		// This should slightly improve the experience of switching between unit types.
+		let newValue;
+
+		if ( [ 'em', 'rem' ].includes( newUnit ) && unit === 'px' ) {
+			// Convert pixel value to an approximate of the new unit, assuming a root size of 16px.
+			newValue = ( quantity / 16 ).toFixed( 2 ) + newUnit;
+		} else if ( [ 'em', 'rem' ].includes( unit ) && newUnit === 'px' ) {
+			// Convert to pixel value assuming a root size of 16px.
+			newValue = Math.round( quantity * 16 ) + newUnit;
+		} else if (
+			[ 'vh', 'vw', '%' ].includes( newUnit ) &&
+			quantity > 100
+		) {
+			// When converting to `vh`, `vw`, or `%` units, cap the new value at 100.
+			newValue = 100 + newUnit;
+		}
+
+		onChange( {
+			...layout,
+			minimumColumnWidth: newValue,
+		} );
+	};
+
+	return (
+		<fieldset>
+			<BaseControl.VisualLabel as="legend">
+				{ __( 'Minimum column width' ) }
+			</BaseControl.VisualLabel>
+			<Flex>
+				<FlexItem isBlock>
+					<UnitControl
+						size={ '__unstable-large' }
+						onChange={ ( newValue ) => {
+							onChange( {
+								...layout,
+								minimumColumnWidth: newValue,
+							} );
+						} }
+						onUnitChange={ handleUnitChange }
+						value={ value }
+						min={ 0 }
+					/>
+				</FlexItem>
+				<FlexItem isBlock>
+					<RangeControl
```

Comment:
Tiny nit: at the left-most position, the thumb of the range control bumps up against the UnitControl:

<img width="278" alt="image" src="https://user-images.githubusercontent.com/14988353/227059814-91473ea4-dda3-4130-8486-a784594bd7a6.png">

In the [HeightControl component](https://github.com/WordPress/gutenberg/blob/79dfbe73c8c8a54ca4576a6570db489effafadbb/packages/block-editor/src/components/height-control/index.js#L114), the `RangeControl` is wrapped in a `Spacer` with `marginX` set to 2, to give it a bit more space. Is it worth including that here, too?

Reply:
I tweaked the gap of the `Flex` component instead, let me know what you think!

Reply:
That's looking better! One extremely nit-picky issue with using gap instead of margin on the range control is that it looks like in the right-most position the Tooltip gets cut-off very slightly:

![2023-03-23 15 53 04](https://user-images.githubusercontent.com/14988353/227106899-35c084b7-065f-46ee-8049-664a7625df52.gif)

This is so minor, though, I doubt anyone else would ever notice, so I don't feel strongly either way. The gap in between looks good, though!

Reply:
I think it's fine, and this way the spacing is to the right of the range is the same as to the left of the unit control. 

I suppose we could look into tweaking the position of the circle on the range control so it doesn't overflow its line as much, but I'm not sure it's really needed.

Reply:
Sounds good, I'll resolve this thread 👍